File: dotnet/src/MyDotey.SCF.Simple/Type/String/StringToBooleanConverter.cs
using System;

namespace MyDotey.SCF.Type.String
{
    /**
    * @author koqizhao
    *
    * May 21, 2018
    */
    public class StringToBooleanConverter : StringConverter<bool?>
    {
        public static readonly StringToBooleanConverter DEFAULT = new StringToBooleanConverter();

        public override bool? Convert(string source) {
            if (string.IsNullOrWhiteSpace(source))
                return null;

            return bool.Parse(source);
        }
    }
}

File: java/scf-simple/src/main/java/org/mydotey/scf/type/string/StringToBooleanConverter.java
package org.mydotey.scf.type.string;

/**
 * @author koqizhao
 *
 * May 21, 2018
 */
public class StringToBooleanConverter extends StringConverter<Boolean> {

    public static final StringToBooleanConverter DEFAULT = new StringToBooleanConverter();

    public StringToBooleanConverter() {
        super(Boolean.class);
    }

    @Override
    public Boolean convert(String source) {
        return Boolean.parseBoolean(source);
    }

}
